### ISSUE TITLE
global symbol issues linking with -fvisibility=hidden on SunOS

### ISSUE THREAD

**Issue by risto3** [LudovicRousseau/PCSC#4]
When building 1.8.16 under SunOS 5.11 (illumos distro), if I don't filter out  ```-fvisibility=hidden```
I get the following linker error with:
```
  CCLD     testpcsc
Undefined			first referenced
 symbol  			    in file
SCardListReaders                    testpcsc.o
SCardSetAttrib                      testpcsc.o
SCardGetAttrib                      testpcsc.o
g_rgSCardRawPci                     testpcsc.o
SCardReconnect                      testpcsc.o
SCardFreeMemory                     testpcsc.o
SCardDisconnect                     testpcsc.o
SCardReleaseContext                 testpcsc.o
SCardIsValidContext                 testpcsc.o
SCardEstablishContext               testpcsc.o
SCardGetStatusChange                testpcsc.o
SCardListReaderGroups               testpcsc.o
SCardStatus                         testpcsc.o
pcsc_stringify_error                testpcsc.o
SCardControl                        testpcsc.o
SCardConnect                        testpcsc.o
g_rgSCardT1Pci                      testpcsc.o
g_rgSCardT0Pci                      testpcsc.o
SCardTransmit                       testpcsc.o
ld: fatal: symbol referencing errors. No output written to .libs/testpcsc
```
I'm running on  pkgsrc SunOS i386 with gcc49 and native ld.
BTW, I have more or less the same issue with ccid.

**Comment by LudovicRousseau** [comment]
Can you propose a patch?
Maybe you should have a look at the file misc.h https://github.com/LudovicRousseau/PCSC/blob/master/src/misc.h

**Comment by risto3** [comment]
ahh, misc.h looks like the first guard is defeating its purpose (when not using studio)
I'll try to check things this week-end... (merci pour l'astuce)

**Comment by risto3** [comment]
I have a patch for misc.h in my tree, I'll see if I can get a pull request working, if not it is here:
https://github.com/risto3/PCSC/commit/2176d782188a9ff4453ae24556a340b2c8f549cf

**Comment by risto3** [comment]
BTW, this same patch to misc.h is needed for ccid...
Can I presume once committed it will fix both pcsc-lite and ccid?

**Comment by LudovicRousseau** [comment]
Yes, the patch will also fix CCID.

I don't see why you can't create a PR. That should be easy using the github web inteface.

**Comment by risto3** [comment]
I just now understood you meant 'Pull Request' and not 'Problem Report'...

I can't seem to get 'create Pull request' to work... perhaps due to the fact I'm
still on Firefox esr24... sorry.

btw, I updated afterwards to put back the gcc3.3 part of the guard that I inadvertently dropped.

**Comment by LudovicRousseau** [comment]
Can you provide a clean and tested Pull Request please?

I also fixed the other SunOS compilation issues. So please rebase your code on my upstream version.

**Comment by risto3** [comment]
create for me is still a no-op, but the diff works
https://github.com/LudovicRousseau/PCSC/compare/master...risto3:sun-visibility

**Comment by LudovicRousseau** [comment]
Fixed in 687128ad9a377a27bd93e542fc096530534dc101
Thanks
